RealmSwift/Logger.swift

```diff
@@ -0,0 +1,346 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Realm
+import Realm.Private
+
+/**
+ Global logger class used by all Realm components.
+
+ Set the global log level for a given category.
+ ```swift
+ Logger.setLogLevel(.info, for: Category.sdk)
+ ```
+
+ Read the global log level for a given category.
+ ```swift
+ let level = Logger.logLevel(for: Category.Storage.all)
+ ```
+
+ You can define your own custom logger creating an instance of `Logger` and defining the log function which will be
+ invoked whenever there is a log message.
+
+ ```swift
+ let logger = Logger(function: { level, category, message in
+    print("Realm Log - \(category.rawValue)-\(level): \(message)")
+ })
+ ```
+
+ Set this custom logger as you default logger using `Logger.shared`. This will replace the default logger.
+
+ ```swift
+ Logger.shared = logger
+ ```
+
+ - note: By default log threshold level is `.info`, for the log category `.Category.realm`,
+         and logging strings are output to Apple System Logger.
+ - SeeAlso: `LogCategory`
+*/
+public typealias Logger = RLMLogger
+extension Logger {
+    /**
+     Log a message to the supplied level.
+
+     ```swift
+     let logger = Logger(level: .info, logFunction: { level, message in
+         print("Realm Log - \(level): \(message)")
+     })
+     logger.log(level: .info, message: "Info DB: Database opened succesfully")
+     ```
+
+     - parameter level: The log level for the message.
+     - parameter category: The log category for the message.
+     - parameter message: The message to log.
+     */
+    internal func log(level: LogLevel, category: LogCategory = Category.realm, message: String) {
```

Comment:
Should the default category here be `Category.sdk` since that method should supposedly only be used to log things from the SDK?